spring-integration-gemfire/src/main/java/org/springframework/integration/gemfire/metadata/GemfireMetadataStore.java

```diff
@@ -82,4 +94,47 @@ public String remove(String key) {
 		return this.region.remove(key);
 	}
 
+	@Override
+	public void addListener(MetadataStoreListener listener) {
+		Assert.notNull(listener, "'listener' must not be null");
+		this.listeners.add(listener);
+	}
+
+	@Override
+	public void removeListener(MetadataStoreListener listener) {
+		this.listeners.remove(listener);
+	}
+
+	private class GemfireCacheListener extends CacheListenerAdapter<String, String> {
```

Comment:
Since we don't have any references to the outer class unless `listeners` I would prefer to make it as `static` and propagate the `GemfireMetadataStore.this.listeners` to the local property of this class.
In addition, please, add package protected constructor to avoid some extra generation around default `private` one. See `MetadataStoreListenerInvokingPathChildrenCacheListener` for example.

Reply:
Addressed in 0b3481c.